web/packages/shared/utils/highbar.ts

```diff
@@ -0,0 +1,273 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export function mergeDeep(target, ...sources) {
+  const isObject = obj => obj && typeof obj === 'object' && !Array.isArray(obj);
+
+  if (!sources.length) return target;
+  const source = sources.shift();
+
+  if (isObject(target) && isObject(source)) {
+    for (const key in source) {
+      if (isObject(source[key])) {
+        if (!target[key]) Object.assign(target, { [key]: {} });
+        mergeDeep(target[key], source[key]);
+      } else {
+        Object.assign(target, { [key]: source[key] });
+      }
+    }
+  }
+
+  return mergeDeep(target, ...sources);
+}
+
+export function isInteger(checkVal: any): boolean {
+  return Number.isInteger(checkVal) || checkVal == parseInt(checkVal);
+}
+
+export function isObject(checkVal) {
+  const type = typeof checkVal;
+  return checkVal != null && (type == 'object' || type == 'function');
+}
+
+// Lift & Shift from lodash
```

Comment:
what does `lift and shift` mean? if we copied from lodash, should we specify the version we copied it from?

Reply:
Lift & Shift: To move something from one position to another with minimal changes.